### DISCUSSION TITLE
Failed to connect with build server

### DISCUSSION THREAD

**Discussion by gmenih** [scalameta/nvim-metals#632] [General]
I'm having some trouble getting nvim-metals to run properly on my M3 MacBook. I've had it running well on my Linux machine, and I'm using the exact same setup. 

When I open a Scala file, I see this error: `LSP[metals] Failed to connect with build server, no functionality will work. See logs for more details.`

Looking at the `.metals/metals.log`, I can see a lot of errors where metals is trying to execute java on path `/opt/homebrew/opt/openjdk@17/bin/bin/java`. This is nothing like my `JAVA_HOME` path, which is set to `/opt/homebrew/Cellar/openjdk/21.0.1/libexec/openjdk.jdk/Contents/Home/`.

Somehow it built a completely wrong path with a `/bin/bin/java`, which doesn't exist (`/bin/java` does). 

I'm not sure what I'm doing wrong at this point. I didn't have any issues setting this up on Linux, but I can't get it working correctly on macOS.

**Comment by ckipp01** [comment]
Hey @gmenih, that's odd that it's picking up the incorrect `JAVA_HOME`. It uses a few different mechanisms to find that, but if you have `JAVA_HOME` set it should for sure be grabbing that. One thing you can try is to actually set this directly in your config settings by setting `javaHome`. Then try opening again and ensuring that this works. Also do you know what version of Metals you're using? You should be able to do a `:MetalsInfo` and get this info.

**Reply by gmenih** [reply]
Okay, I figured out the issue. It looks like I did at some point have an incorrect `JAVA_HOME` set, which was then baked into `~/.bloop/bloop.json` and metals was using that variable.

After deleting `~/.bloop`, it is working correctly!
